### PR TITLE
Premint - premint executor is upgradeable

### PR DESCRIPTION
**Pull request by oveddan**
* Premint Executor contract is now upgradeable, using using Ownable2StepUpgradeable, UUPSUpgradeable
* Moved some common test fixture setup functionality to libs

### CODE REVIEW

**Comment by oveddan** [comment]
Current dependencies on/for this PR:
* main
  * **PR #161** <a href="https://app.graphite.dev/github/pr/ourzora/zora-1155-contracts/161" target="_blank">Graphite</a> 
    * **PR #166** <a href="https://app.graphite.dev/github/pr/ourzora/zora-1155-contracts/166" target="_blank">Graphite</a> 
    * **PR #164** <a href="https://app.graphite.dev/github/pr/ourzora/zora-1155-contracts/164" target="_blank">Graphite</a> 
    * **PR #163** <a href="https://app.graphite.dev/github/pr/ourzora/zora-1155-contracts/163" target="_blank">Graphite</a> 
      * **PR #168** <a href="https://app.graphite.dev/github/pr/ourzora/zora-1155-contracts/168" target="_blank">Graphite</a>  👈
    * **PR #162** <a href="https://app.graphite.dev/github/pr/ourzora/zora-1155-contracts/162" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ourzora/zora-1155-contracts/168?utm_source=stack-comment).

File: src/premint/ZoraCreator1155PremintExecutor.sol [hunk 1]
```diff
@@ -162,4 +168,27 @@ contract ZoraCreator1155PremintExecutor {
             isValid = IZoraCreator1155(contractAddress).isAdminOrRole(recoveredSigner, CONTRACT_BASE_ID, PERMISSION_BIT_MINTER);
         }
     }
+
+    // upgrade related functionality
+
+    /// @notice The name of the contract for upgrade purposes
+    function contractName() external pure returns (string memory) {
```
Comment (kulkarohan) [inline src/premint/ZoraCreator1155PremintExecutor.sol:175]: thoughts on having this be `contractVersion` instead ? @oveddan @iainnash

Comment (oveddan) [reply src/premint/ZoraCreator1155PremintExecutor.sol:175]: we can also add a contract version - this is mainly used when upgrading, to make sure the contract upgrading to has the same `contractName` as this one, that wouldn't quite make sense in the context of versions, but there could be another check that ensures the version is increasing.

Comment (oveddan) [reply src/premint/ZoraCreator1155PremintExecutor.sol:175]: i think actually adding the contract version makes sense too, just so we have it on record

Comment (kulkarohan) [reply src/premint/ZoraCreator1155PremintExecutor.sol:175]: ah i see yeah lets keep

**Comment by kulkarohan** [comment]
if the libraries are causing issues with verification and events, we can get away with using a single contract for the premint executor since its standalone. 

but if those aren't issues then this lgtm

**Review by kulkarohan — APPROVED** [review]

**Comment by oveddan** [comment]
> if the libraries are causing issues with verification and events, we can get away with using a single contract for the premint executor since its standalone. 
> 
> but if those aren't issues then this lgtm

totally agree would be good to move from library to interface esp cause of verification. will continue looking into it as part of #166

**Pull request merged**